library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/net/BatchesToSegmentsMapper.kt

```diff
@@ -42,74 +48,96 @@ class BatchesToSegmentsMapper {
                     null
                 }
             }
-            .map {
-                val applicationId = it.get(EnrichedRecord.APPLICATION_ID_KEY).asString
-                val sessionId = it.get(EnrichedRecord.SESSION_ID_KEY).asString
-                val viewId = it.get(EnrichedRecord.VIEW_ID_KEY).asString
-                val context = SessionReplayRumContext(applicationId, sessionId, viewId)
-                val records = it.get(EnrichedRecord.RECORDS_KEY).asJsonArray
-                Pair(context, records)
-            }
-            .groupBy { it.first }
-            .mapValues {
-                it.value.fold(JsonArray()) { acc, pair ->
-                    acc.addAll(pair.second)
-                    acc
-                }
-            }
-            .filter { !it.value.isEmpty }
-            .mapNotNull { entry ->
-                @Suppress("SwallowedException")
-                try {
-                    groupToSegmentsPair(entry)
-                } catch (e: JsonParseException) {
-                    // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-                    null
-                } catch (e: IllegalStateException) {
-                    // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
+            .reduceOrNull { accumulator, pair ->
+                val records = accumulator.second
+                val newRecords = pair.second
+                records.addAll(newRecords)
+                Pair(accumulator.first, records)
+            } ?: return null
+
+        return mapToSegment(reducedEnrichedRecord.first, reducedEnrichedRecord.second)
+    }
+
+    private fun mapToSegment(rumContext: SessionReplayRumContext, records: JsonArray):
```

Comment:
shouldn't operations with json in this method be in the `try-catch` as well? like in `groupBatchDataIntoSegments` method. Or maybe we can extract everything in the typed object in one place and then continue working with it.

Reply:
Good catch, not sure extracting it will work as I still need to modify the serialized segment by adding the ordered records.